### PR TITLE
Clarify behavior of max_shards_per_node and total_shards_per_node

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Source: https://github.com/elastic/elasticsearch/commit/e4054e4e926afbbf30ad7cca557d0a67d6126448 and it's [issue](https://github.com/elastic/elasticsearch/issues/51839)

and https://github.com/elastic/elasticsearch/commit/eb3d18479e3ec756351a50a49b87ae2f71673a9c

and https://www.elastic.co/guide/en/elasticsearch/reference/7.17/index-modules.html (index.auto_expand_replicas)

Closes https://github.com/crate/crate/issues/15803. See https://github.com/crate/crate/issues/15803#issuecomment-2058729661

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
Table creation specific: Debugged locally, `numberOfReplicas` in `ShardLimitValidator` (creation, `max_shards_per_node`) is 0 if replicas are given as a range.

**Comment by BaurzhanSakhariev** [comment]
Alternative can be diverging from ES and tweaking `ShardLimitValidator` to take to account auto expanding replicas and throw an error. 

Example of a small limit and big number of shards due to CREATE TABLE with replicas provided as a range. https://github.com/crate/crate/commit/fb67ea6d571c08046d438e8f5810610b441830c7#diff-0cd3590e9880e5b3ed3bb81d1a7020cfff0324787eab6c88b29717e9155d3aecR85-R102

File: docs/sql/statements/create-table.rst [hunk 1]
```diff
@@ -386,6 +386,13 @@ The number of replicas is defined like this::
   the number of data nodes in the cluster. If ``max_replicas`` is the string
   ``all`` then it will always be N-1.
 
+.. NOTE::
+
+   If value provided as a range or not provided and default ``0-1`` is used,
```
[the remark this answers is not in the document]
Comment (BaurzhanSakhariev) [reply docs/sql/statements/create-table.rst:391]: yes. Range is not checked. Default is range and thus we don't check the limit for default value as well.

**Comment by BaurzhanSakhariev** [comment]
Hi @romseygeek, thanks for the suggestions! 

I rephrased the note in `CREATE TABLE`  and adjusted other changes to fit into 80 symbols per line. 
Could you take another look?

File: docs/sql/statements/create-table.rst [hunk 2]
```diff
@@ -386,6 +386,13 @@ The number of replicas is defined like this::
   the number of data nodes in the cluster. If ``max_replicas`` is the string
   ``all`` then it will always be N-1.
 
+.. NOTE::
+
+   If value is provided as a range or default value ``0-1`` is used,
```
Comment (romseygeek) [inline docs/sql/statements/create-table.rst:391]:
```suggestion
   If the value is provided as a range or the default value ``0-1`` is used,
```

**Review by romseygeek — APPROVED** [review]
One more grammar nit, LGTM otherwise.

**Comment by seut** [comment]
@BaurzhanSakhariev I'm a bit confused. Your new changes states: 

`If the value is provided as a range or the default value 0-1 is used, [cluster.c](https://cratedb.com/docs/crate/reference/en/master/config/cluster.html#cluster-max-shards-per-node) and [cluster.routing.allocation.total_shards_per_node](https://cratedb.com/docs/crate/reference/en/master/config/cluster.html#cluster-routing-allocation-total-shards-per-node) limits are not checked.`

From looking up the code, this sounds wrong. The possible expanded replicas won't be taken into account, while the number of shards is indeed taken into account.

**Comment by BaurzhanSakhariev** [comment]
> From looking up the code, this sounds wrong. The possible expanded replicas won't be taken into account, while the number of shards is indeed taken into account and checked.

Good point, create https://github.com/crate/crate/pull/15887 to update the docs

**Pull request merged**